src/org/traccar/database/ConnectionManager.java

```diff
@@ -142,26 +146,22 @@ public void run(Timeout timeout) throws Exception {
         updateDevice(device);
     }
 
-    public Event updateDeviceState(long deviceId) {
+    public Set<Event> updateDeviceState(long deviceId) {
         DeviceState deviceState = Context.getDeviceManager().getDeviceState(deviceId);
-        if (deviceState == null || deviceState.getMotionState() == null) {
-            return null;
+        Set<Event> result = new HashSet<>();
+
+        Event event = MotionEventHandler.updateMotionState(deviceState, tripsConfig);
+        if (event != null) {
+            result.add(event);
         }
-        Event result = null;
-        Boolean oldMotion = deviceState.getMotionState();
-        long currentTime = System.currentTimeMillis();
-        boolean newMotion = !oldMotion;
-        Position motionPosition = deviceState.getMotionPosition();
-        if (motionPosition != null) {
-            long motionTime = motionPosition.getFixTime().getTime()
-                    + (newMotion ? tripsConfig.getMinimalTripDuration() : tripsConfig.getMinimalParkingDuration());
-            if (motionTime <= currentTime) {
-                String eventType = newMotion ? Event.TYPE_DEVICE_MOVING : Event.TYPE_DEVICE_STOPPED;
-                result = new Event(eventType, motionPosition.getDeviceId(), motionPosition.getId());
-                deviceState.setMotionState(newMotion);
-                deviceState.setMotionPosition(null);
-            }
+
+        event = OverspeedEventHandler.updateOverspeedState(deviceState, Context.getDeviceManager().
+                lookupAttributeDouble(deviceId, OverspeedEventHandler.ATTRIBUTE_SPEED_LIMIT, 0, false),
```

Comment:
What if we make `updateXxx` functions part of `DeviceState` class?
Agree, initializing parameters looks mess, because it is shared, may be we place them in `Context` ?